icaruscode/Filters/FilterCRTPMTMatching_module.cc

```diff
@@ -378,202 +200,82 @@ bool icarus::crt::FilterCRTPMTMatching::filter(art::Event& e) {
-                                   exiting.CRTHit->y_pos,
-                                   exiting.CRTHit->z_pos};
-          double CRTtime = exiting.CRTHit->ts1_ns / 1e3;
-          int CRTRegion = exiting.CRTHit->plane;
-          int CRTSys = 0;
-          if (CRTRegion >= 36) CRTSys = 1;
-          double CRTTof_opflash = exiting.CRTHit->ts1_ns - tflash * 1e3;
-          std::vector<int> HitsFebs;
-          for (auto crts : exiting.CRTHit->feb_id) {
-            HitsFebs.emplace_back((int)crts);
-          }
-          if (CRTSys == 0) TopEx++;
-          if (CRTSys == 1) SideEx++;
-          MatchedCRT thisCRTMatch = { /* .CRTHitPos = */ thisCRTpos, // C++20: restore initializers
-                                     /* .CRTPMTTimeDiff_ns = */ CRTTof_opflash,
-                                     /* .CRTTime_us = */ CRTtime,
-                                     /* .CRTSys = */ CRTSys,
-                                     /* .CRTRegion = */ CRTRegion};
-          thisFlashCRTmatches.push_back(thisCRTMatch);
-        }
-      }
-      FlashType thisFlashType = { /* .FlashPos = */ flash_pos, // C++20: restore initializers
-                                 /* .FlashTime_us = */ tflash,
-                                 /* .FlashGateTime_ns = */ fThisRelGateTime,
-                                 /* .inBeam = */ fThisInTime_beam,
-                                 /* .inGate = */ inTime,
-                                 /* .Classification = */ eventType,
-                                 /* .CRTmatches = */ thisFlashCRTmatches};
-      if (inTime == true) thisEventFlashes.push_back(thisFlashType);
+  std::vector<CRTPMTMatching> EventFlashes;
+  for(auto const & crtpmt : crtpmtMatches) {
+    if(fSpillOnly==true){
+	if(crtpmt.flashInBeam==1) EventFlashes.push_back(crtpmt);
     }
-  }
+    else {
+	if(crtpmt.flashInGate==1) EventFlashes.push_back(crtpmt);
+    }
+  } 
+
   bool hasOnlyCosmics = false;
-  if (fFilterLevel == "loose")
+  if (fFilterLevel == "loose") {
     hasOnlyCosmics = false;  // By default, with loose Filtering, everything is
-                         // "intersting", nothing tagged as clear cosmic
+                             // "intersting", nothing tagged as clear cosmic
+    if(fSpillOnly==true && EventFlashes.empty()) hasOnlyCosmics = true;
+  }
   else if (fFilterLevel == "medium") {
     hasOnlyCosmics = true;
-    for (const auto& h : thisEventFlashes) {
+    for (const auto& f : EventFlashes) {
       bool isCosmic = false;
-      if (h.Classification == enTop || h.Classification == enTop_exSide ||
-          h.Classification == enTop_mult ||
-          h.Classification == enTop_exSide_mult)
+      if (f.flashClassification == icarus::crt::MatchType::enTop || f.flashClassification == icarus::crt::MatchType::enTop_exSide ||
+          f.flashClassification == icarus::crt::MatchType::enTop_mult ||
+          f.flashClassification == icarus::crt::MatchType::enTop_exSide_mult) {
```

Comment:
Are we making any judgement about tracks entering from the bottom?
A comment may be useful here.

Reply:
i personally think no, because the default is to not perform the matching with bottom CRT hits. i did not make any modifications to the filter module with the modifications to match bottom crt if fcl parameter is true, but maybe we want the option to do matching with the bottom there?